src/mkindex_algo.hpp

```diff
@@ -636,28 +642,17 @@ auto generateIndex(TStringSet                       & seqs,
                    LambdaIndexerOptions       const & options)
 {
     using TRedAlph       = seqan3::range_innermost_value_t<TStringSet>;
-
-    constexpr auto is_collection = seqan3::text_layout::collection;
-    using TSpec = IndexSpec<seqan3::alphabet_size<TRedAlph>>;
-    using TIndex = std::conditional_t<is_bi,
-                                      seqan3::bi_fm_index<TRedAlph, is_collection, TSpec>,
-                                      seqan3::fm_index<TRedAlph, is_collection, TSpec>>;
+    using TIndexSpec     = IndexSpec<seqan3::alphabet_size<TRedAlph>>;
+    using TIndex         = std::conditional_t<is_bi,
+                                             fmindex_collection::BiFMIndex<TIndexSpec>,
+                                             fmindex_collection::ReverseFMIndex<TIndexSpec>>;
 
     myPrint(options, 1, "Generating Index...");
     double s = sysTime();
 
-    TIndex index;
-
-    if constexpr (is_bi)
-    {
-        // WORKAROUND https://github.com/seqan/seqan3/pull/1519
-        std::vector<std::vector<TRedAlph>> tmp = seqs | seqan3::views::to<std::vector<std::vector<TRedAlph>>>;
-        index = TIndex{tmp};
-    }
-    else
-    {
-        index = TIndex{seqs};
-    }
+    // creates the index with sampling rate of 5
+    TIndex index{seqs | std::views::transform([] (auto const& seq) { return seq | seqan3::views::to_rank; })
```

Comment:
```suggestion
    TIndex index{seqs | seqan3::views::to_rank
```

This should be enough, because the view is a "deep" view.